packages/analytics/amplify_analytics_pinpoint_dart/lib/src/analytics_plugin_impl.dart

```diff
@@ -153,11 +153,14 @@ class AmplifyAnalyticsPinpointDart extends AnalyticsPluginInterface {
       },
     );
 
-    _autoEventSubmitter = StoppableTimer(
-      duration: const Duration(seconds: 10),
-      callback: flushEvents,
-      onError: (e) => _logger.warn('Exception in events auto flush', e),
-    );
+    final autoFlushInterval = pinpointConfig.autoFlushEventsInterval;
+    if (autoFlushInterval.inSeconds > 0 || !autoFlushInterval.isNegative) {
```

Comment:
Huh? These conditions check the same thing, no? And shouldn't this use the default if it's invalid. This would mean it never auto flushes and the user gets no indication if I'm reading this correctly. Better to throw a `ConfigurationError`, I think, or print a warning and fallback to default.